cmake/multi_image.cmake

```diff
@@ -232,6 +232,12 @@ function(add_child_image_from_source)
     list(APPEND extra_cmake_args "-DCONFIG_NCS_MCUBOOT_IN_BUILD=y")
   endif()
 
+  if (CONFIG_SECURE_BOOT)
+    # If secure boot is enabled, propagate the firmware version to child
+    # images from the parent
+    list(APPEND extra_cmake_args "-DCONFIG_FW_INFO_FIRMWARE_VERSION=${CONFIG_FW_INFO_FIRMWARE_VERSION}")
```

Comment:
I wonder if this is the correct approach. User may still configure the s1_image separately from menuconfig which would lead to unexpected behavior. 

Maybe we should simply disallow configuring s1_image separately to ensure that configuration is the same for S0 and S1 images. Otherwise we may need to take into account S1 image version while generating manifest.json in dfu_application.zip.

Reply:
IMO this is not what we want, the child image should not have the same firmware version as the parent image. Could you explain what the motivation behind this change is?

Reply:
Propagating the information to `s1_image1` introduced in this PR seems to contradict with the fact that `the child image should not have the same firmware version as the parent image`.

Reply:
Synchronizing configuration among the S0 and S1 images could mitigate this issue and it would be acceptable approach for me.

Reply:
I also noticed that if I provide the firmware version in the command line command (`west build -b nrf52840dk_nrf52840 --pristine – -DCONFIG_FW_INFO_FIRMWARE_VERSION=2`) image version for both image slots is updated on main. If I do the same command on 5ae1c7619db45f612fbf022ab5f244f007bc35f0, only S0 version seems updated (checked with nRF desktop application).

Either it's a race or there used to be issue that was fixed (I did not see any related fix between main and mentioned commit, but maybe I missed something). @nordicjm, @hakonfam, do you know what could cause the observed behavior?

Reply:
> Synchronizing configuration among the S0 and S1 images could mitigate this issue and it would be acceptable approach for me (both images should have the same version in that case).

That is the correct solution, been trying to get that working but without success, have pinged @tejlmand earlier for assistance. That's why I've marked this as DNM, it needs rearchitecting to fix the underlying issue, the configuration of s1 should always match s0 with the addition of the variant tag.

Reply:
> Either it's a race or there used to be issue that was fixed (I did not see any related fix between main and mentioned commit, but maybe I missed something). @nordicjm, @hakonfam, do you know what could cause the observed behavior?

Hakon said that previously, the image would be built once and linked twice, one for s0 and one for s1, so the configuration for those would be identical. At some point, it has changed from building once/linking twice to building twice, from that change it will no longer synchronise local changes made to s0 on to the s1 image, the original configuration will be applied to both however

Reply:
Changing approach to building twice was done quite some time ago already. The mentioned commit hash is not that old.
If you are sure that configurations applied as additional CMake options would be properly applied to both images then I am fine with the change :)